packages/roosterjs-content-model-plugins/lib/autoFormat/numbers/transformOrdinals.ts

```diff
@@ -16,16 +16,15 @@ const getOrdinal = (value: number) => {
 
 /**
  * @internal
- */
-export function transformOrdinals(
+ */ export function transformOrdinals(
     previousSegment: ContentModelText,
     paragraph: ShallowMutableContentModelParagraph,
     context: FormatContentModelContext
 ): boolean {
     const value = previousSegment.text.split(' ').pop()?.trim();
     if (value) {
         const ordinal = value.substring(value.length - 2);
-        const ordinalValue = parseInt(value);
+        const ordinalValue = getValue(value);
```

Comment:
You can pass in `ordinal` instead, so no need to substring again

Reply:
One substring is to get the st, nd, rd, and th and the other is for the numeric part.